numba/tests/test_np_functions.py

```diff
@@ -2503,6 +2507,120 @@ def check_pass_through(jitted, expect_same, params):
+                               itertools.product(target_numpy_values,
+                                                 target_numpy_types))
+
+        target_non_numpy_inputs = [
+            1,
+            1.0,
+            True,
+            1j,
+            [0, 1, 2],
+            (0, 1, 2),
+        ]
+        for i in itertools.chain(target_numpy_inputs, target_non_numpy_inputs):
+            check(i, repeats=0)
+            check(i, repeats=1)
+            check(i, repeats=2)
+            check(i, repeats=3)
+            check(i, repeats=100)
+
+        # check broadcasting when repeats is an array
+        one = np.arange(1)
+        check(one, repeats=np.array([0]))
+        check(one, repeats=np.array([1]))
+        check(one, repeats=np.array([2]))
+
+        two = np.arange(2)
+        check(two, repeats=np.array([0, 0]))
+        check(two, repeats=np.array([0, 1]))
+        check(two, repeats=np.array([1, 0]))
+        check(two, repeats=np.array([1, 1]))
+
+        check(two, repeats=np.array([1, 2]))
+        check(two, repeats=np.array([2, 1]))
+        check(two, repeats=np.array([2, 2]))
+        check(two, repeats=np.array([2, 2], dtype=np.int32))
+
+        check(np.arange(10), repeats=np.arange(10))
+
+    def test_repeat_exception(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+        self.disable_leak_check()
+
+        # heterogeneous list
+        with self.assertRaises(TypeError) as e:
+            cfunc([0, 1.0], 1)
+        self.assertIn("can't unbox heterogeneous list: int64 != float64",
+                      str(e.exception))
+
+        # nested list
+        with self.assertRaises(TypeError) as e:
+            cfunc([[0, 1], [2, 3], 1.0], 1)
+        self.assertIn("cannot reflect element of reflected container: "
+                      "reflected list(reflected list(int64))",
+                      str(e.exception))
+
+        # negative repeat argument
+        with self.assertRaises(ValueError) as e:
+            cfunc(np.ones(1), -1)
+        self.assertIn("negative dimensions are not allowed",
+                      str(e.exception))
+
+        # negative repeat argument as array
+        with self.assertRaises(ValueError) as e:
+            cfunc(np.ones(2), np.array([1, -1]))
+        self.assertIn("count < 0",
+                      str(e.exception))
+
+        # broadcasting error, repeats too large
+        with self.assertRaises(ValueError) as e:
+            cfunc(np.ones(2), np.array([1, 1, 1]))
+        self.assertIn("operands could not be broadcast together",
+                      str(e.exception))
+
+        # broadcasting error, repeats too small
+        with self.assertRaises(ValueError) as e:
+            cfunc(np.ones(5), np.array([1, 1, 1, 1]))
+        self.assertIn("operands could not be broadcast together",
+                      str(e.exception))
+
+        with self.assertRaises(TypingError):
```

Comment:
Think this will exit the context managed block on the first exception, not sure that's intended? Perhaps re-write as a loop like:
```python
for rep in [True, 1.0, "a", [1, 1],]: # (don't need "a" and "1", both are UnicodeType)
  with self.assertRaises(TypingError):
    cfunc(np.ones(1), rep)
```

Reply:
This sounds like a good idea, I am going to leave the "1" in, because that should work as it works in numpy, but there is no way for us to cast that to an int in numba.